rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java

```diff
@@ -0,0 +1,126 @@
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+@Slf4j
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class XsltTransformation extends Recipe {
+
+    @Nullable
+    @Language("xml")
+    @Option(displayName = "XSLT Configuration transformation",
+            description = "The transformation to be applied.")
+    String xslt;
+
+    @Nullable
+    @Option(displayName = "XSLT Configuration transformation classpath resource",
+            description = "Recipe transformation provided as a classpath resource.")
+    String xsltResource;
+
+    @Option(displayName = "File pattern",
+            description = "A glob expression that can be used to constrain which directories or source files should be searched. " +
+                          "Multiple patterns may be specified, separated by a semicolon `;`. " +
+                          "If multiple patterns are supplied any of the patterns matching will be interpreted as a match.",
+            example = "**/*.xml")
+    String filePattern;
+
+    @Override
+    public String getDisplayName() {
+        return "XSLT transformation";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Apply the specified XSLT transformation on.";
+    }
+
+    @JsonCreator
+    public XsltTransformation(
+            @Nullable @JsonProperty("xslt") @Language("xml") String xslt,
+            @Nullable @JsonProperty("xsltResource") String xsltResource,
+            @JsonProperty("filePattern") String filePattern
+    ) {
+        this.xslt = xslt;
+        this.xsltResource = xsltResource;
+        this.filePattern = filePattern;
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        TreeVisitor<?, ExecutionContext> visitor = new XsltTransformationVisitor(loadResource(xslt, xsltResource));
+
+        @SuppressWarnings("unchecked")
+        TreeVisitor<?, ExecutionContext> check = Preconditions.or(Arrays.stream(filePattern.split(";"))
+                .map(HasSourcePath<ExecutionContext>::new)
+                .toArray(TreeVisitor[]::new));
+
+        visitor = Preconditions.check(check, visitor);
+
+        return visitor;
+    }
+
+    @Override
+    public Validated<Object> validate() {
+        Validated<Object> validated = super.validate();
+
+        if (StringUtils.isBlank(xslt) && StringUtils.isBlank(xsltResource)) {
+            validated = validated.and(Validated.missing("xslt", xslt, "XSLT")
+                    .or(Validated.missing("xsltResource", xsltResource, "XSLT Classpath resource")));
+        }
+
+        if (!StringUtils.isBlank(xslt) && !StringUtils.isBlank(xsltResource)) {
+            validated = validated.and(Validated.missing("xslt", xslt, "XSLT")
+                    .and(Validated.missing("xsltResource", xsltResource, "XSLT Classpath resource")));
```

Comment:
Do the validated method calls here not imply the calls to `StringUtils.isBlank` ? I'd be quite surprised if we need to check these explicitly, but I'm not in my IDE as I go through this first review pass.

Reply:
The idea here was to not let the user specify both `xslt` and `xsltResource` at the same time.